torch/autograd/gradcheck.py

```diff
@@ -61,6 +61,24 @@ def get_numerical_jacobian(fn, input, target=None, eps=1e-3):
     x_tensors = iter_tensors(target, True)
     j_tensors = iter_tensors(jacobian)
 
+    def compute_gradient(x, idx, is_mkldnn=False):
+
+        def fn_out(is_mkldnn=False):
+            if not is_mkldnn:
+                return fn(input).clone()
+            else:
+                return fn([x.to_mkldnn()])
+
+        orig = x[idx].item()
+        x[idx] = orig - eps
+        outa = fn_out(is_mkldnn)
+        x[idx] = orig + eps
+        outb = fn_out(is_mkldnn)
+        if not is_mkldnn:
```

Comment:
I think this is fairly confusing if you don't see the rest of the code from the mkldnn part. Because it would suggest we don't write back the original value.

Reply:
I think it's also ok if we remove this check. we don't have to set back the orig value for x in case of mkldnn since we are not overwriting the x_tensor in the code below but it's also ok to set it back.